saleor/graphql/product/bulk_mutations/products.py

```diff
@@ -431,6 +431,9 @@ def save_variants(cls, info, instances, product, cleaned_inputs):
             cls.save(info, instance, cleaned_input)
             cls.create_variant_stocks(instance, cleaned_input)
             cls.create_variant_channel_listings(instance, cleaned_input)
+            transaction.on_commit(
```

Comment:
This the case that I explained to you on the call.  You're calling `lambda` with an instance which you get by iterating on something. `On commit` will be called only with the last `instance` object.